src/script/cpp_api/s_base.h

```diff
@@ -71,7 +76,11 @@ class ServerActiveObject;
 
 class ScriptApiBase {
 public:
-	ScriptApiBase();
+	ScriptApiBase(ScriptingType  type);
+	ScriptApiBase()
+	{
+		FATAL_ERROR_IF(true, "ScriptApiBase created without ScriptingType!");
```

Comment:
This can just be `FATAL_ERROR`.  Also, is this really necessary?  If so, please add a comment explaining why.